File: crates/enforcer-memory/tests/fixtures/memory/lang_csharp/Widget.cs
using System;

namespace WidgetApp
{
    public interface IDrawable
    {
        string Draw();
    }

    public class BaseWidget
    {
        public const int MaxWidgets = 10;

        public virtual string Describe()
        {
            return "base";
        }
    }

    public class Widget : BaseWidget, IDrawable
    {
        public string Name;

        public Widget(string name)
        {
            Name = name;
        }

        public string Draw()
        {
            return LoadWidgetSettings(Name);
        }

        private string LoadWidgetSettings(string name)
        {
            return name;
        }
    }
}

File: crates/enforcer-memory/tests/fixtures/memory/lang_php/Widget.php
<?php

namespace WidgetApp;

interface Drawable
{
    public function draw(): string;
}

class BaseWidget
{
    const MAX_WIDGETS = 10;

    public function describe(): string
    {
        return "base";
    }
}

class Widget extends BaseWidget implements Drawable
{
    public string $name;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    public function draw(): string
    {
        return $this->loadWidgetSettings($this->name);
    }

    private function loadWidgetSettings(string $name): string
    {
        return $name;
    }
}
